aarch64_qemuvirt/src/main.rs

```diff
@@ -44,33 +36,28 @@ extern "C" fn k_main(_device_tree_ptr: usize) -> ! {
         asm::barrier::dsb(asm::barrier::SY);
     };
 
-    let device_tree_ptr = DTB_ADDR;
-    let device_tree = kernel::device_tree::DeviceTree::new(device_tree_ptr);
-    //
-    let pmm = kernel::mm::PhysicalMemoryManager::from_device_tree(&device_tree, 4096);
-    let mut mm = kernel::mm::MemoryManager::new(pmm);
-    let pagetable = kernel::mm::map_address_space(
-        &device_tree,
-        &mut mm,
-        &[kernel::kernel_console::get_console()],
-    );
-    mm.set_kernel_pagetable(pagetable);
-
-    // ???:
-    // why do we pass the pagetable after into the mm ?
-    //     (can we pass it at initialization)
-    // - also when we map, we reload the page table, enabling the MMU (SCTLR_EL1.M)
-    //   shouldn't that be separate ? ie map just put it in the pagetable, then we can manually reload it / enable paging ?
+    let device_tree = kernel::device_tree::DeviceTree::new(DTB_ADDR);
 
-    kernel::kprintln!("Kernel has been initialized");
+    kernel::globals::PHYSICAL_MEMORY_MANAGER
+        .lock(|pmm| pmm.init_from_device_tree(&device_tree, 4096));
+    kernel::mm::map_address_space(&device_tree, &[&PL011]);
```

Comment:
Note for later, a `register_driver` or something could be useful to avoid having to craft this array by hand

Reply:
Yes, I have a few plans for that.
That well go in a later PR